File: Programmers/[92341] 주차 요금 계산/KingDonggyu/92341.js
function solution(fees, records) {
  const convertMinute = (time) => {
    const [hour, minute] = time.split(':');
    return +hour * 60 + +minute;
  };

  const calculateFee = (time) => {
    if (fees[0] >= time) {
      return fees[1];
    }
    return fees[1] + Math.ceil((time - fees[0]) / fees[2]) * fees[3];
  };

  const parkingInfo = new Map();

  records.forEach((record) => {
    let [time, num, act] = record.split(' ');

    time = convertMinute(time);

    if (parkingInfo.has(num)) {
      const info = parkingInfo.get(num);
      if (act === 'IN') {
        info.entryTime = time;
        return;
      }
      info.totalTime += time - info.entryTime;
      info.entryTime = null;
      return;
    }

    parkingInfo.set(num, {
      entryTime: time,
      totalTime: 0,
    });
  });

  const answer = [];
  const sortedNums = Array.from(parkingInfo.keys()).sort();

  sortedNums.forEach((num) => {
    let { entryTime, totalTime } = parkingInfo.get(num);

    if (entryTime !== null) {
      totalTime += convertMinute('23:59') - entryTime;
    }

    answer.push(calculateFee(totalTime));
  });

  return answer;
}

File: Programmers/[92341] 주차 요금 계산/Johoseong/92341.py
import math
from decimal import Decimal
def solution(fees, records):
    answer = dict()
    cars = dict()
    last = 23 * 60 + 59

    for r in records: # 입/출차 기록하기
        tmp, num, state = r.split()
        h, m = tmp.split(":")
        time = int(h) * 60 + int(m)
        num = int(num)
        if state == "IN":
            cars.setdefault(num, [])
            cars[num].append([time, last])
        else:
            cars[num][-1][1] = time

    for c in cars: # 주차비용 계산
        time = 0
        for i, o in cars[c]:
            time += (o - i)
        if time > fees[0]:
            tmp = Decimal((time - fees[0]) / fees[2])
            answer[c] = fees[1] + math.ceil(tmp) * fees[3]
        else:
            answer[c] = fees[1]
    answer = [i[1] for i in sorted(answer.items())]
    return answer